tests/repository_simulator.py

```diff
@@ -300,10 +300,12 @@ def update_timestamp(self):
         """
         self.timestamp.snapshot_meta.version = self.snapshot.version
 
+        hashes = None
+        length = None
         if self.compute_metafile_hashes_length:
             hashes, length = self._compute_hashes_and_length("snapshot")
-            self.timestamp.snapshot_meta.hashes = hashes
-            self.timestamp.snapshot_meta.length = length
+        self.timestamp.snapshot_meta.hashes = hashes
+        self.timestamp.snapshot_meta.length = length
 
```

Comment:
This implementation is correct and  it does fix the bug but looking at `update_snapshot`, what do you think about replacing these three lines
```
        self.timestamp.snapshot_meta.version = self.snapshot.version
        self.timestamp.snapshot_meta.hashes = hashes
        self.timestamp.snapshot_meta.length = length
```
by directly creating a new `MetaFile` object instead?